app/preprints/detail/controller.ts

```diff
@@ -78,11 +78,25 @@ export default class PrePrintsDetailController extends Controller {
             DATE_LABEL.created;
     }
 
+    get showEditButton() {
+        const providerIsPremod = this.model.provider.reviewsWorkflow === PreprintProviderReviewsWorkFlow.PRE_MODERATION;
+        const preprintIsRejected = this.model.preprint.reviewsState === ReviewsState.REJECTED;
+        const preprintIsInitialVersion = this.model.preprint.version === 1;
+        if (this.userIsContrib && this.model.preprint.isLatestVersion) {
+            if (providerIsPremod && preprintIsRejected && !preprintIsInitialVersion) {
+                return false;
+            }
+            return true;
+        }
+        return false;
+    }
```

Comment:
During my testing, the only issue is I cannot edit my first pending pre-mod submission, which is consistent with the code.

![Screenshot 2025-01-09 at 13 01 23](https://github.com/user-attachments/assets/7b841d48-afce-4d56-9e8b-c76b3baa09d9)

This is the logic BE has for [`is_edit_allowed`](https://github.com/CenterForOpenScience/osf.io/blob/feature/preprints-doi-versioning/api/preprints/views.py#L78C1-L84C22)

```python
@staticmethod
def is_edit_allowed(preprint):
    return True if any([
        preprint.is_latest_version,
        preprint.machine_state in ['initial', 'rejected'],
        preprint.provider.reviews_workflow == Workflows.PRE_MODERATION.value and
        preprint.machine_state == 'pending',
]) else False
```

Reply:
Ahh, ok, I didn't realize that was an editable state!

Reply:
As discussed, BE has a bug here too, here is the updated permission

```python
@staticmethod
def is_edit_allowed(preprint):
    if preprint.is_latest_version or preprint.machine_state == 'initial':
        return True
    if preprint.provider.reviews_workflow == Workflows.PRE_MODERATION.value:
        if preprint.machine_state == 'pending':
            return True
        if preprint.machine_state == 'rejected' and preprint.version == VersionedGuidMixin.INITIAL_VERSION_NUMBER:
            return True
    return False
```